app/Main.hs

```diff
@@ -12,28 +13,34 @@ main :: IO ()
 main = do
 
     -- General
-    cardanoConfiguration            <-  loadCardanoConfiguration
-    cardanoEnvironment              <-  initializeCardanoEnvironment
+    cardanoConfiguration <- loadCardanoConfiguration
+    cardanoEnvironment   <- initializeCardanoEnvironment
 
     -- We check that the application is not already running.
     -- _ <- checkIfApplicationIsRunning cardanoConfiguration
 
     -- And example of an application that goes haywire.
-    let application :: IO ()
-        application = do
+    let application :: LoggingLayer -> IO ()
+        application ll = do
+            let logTrace   = llStartTrace ll
+                logNotice  = llLogNotice  ll
+                appendName = llAppendName ll
+            logNotice logTrace "Hello from logging layer ..."
+            logTrace' <- appendName "cardano-shell" logTrace
+            logNotice logTrace' "Hello #2 from logging layer ..."
             _ <- replicateM 5 (threadDelay 1000000 >> putTextLn "Running node/wallet/whatever!")
             _ <- throwIO UnknownFailureException
             _ <- replicateM 5 (threadDelay 1000000 >> putTextLn "Running node/wallet/whatever!")
             pure ()
 
-    let cardanoApplication :: CardanoApplication
-        cardanoApplication = CardanoApplication application
+    let cardanoApplication :: LoggingLayer -> CardanoApplication
+        cardanoApplication = CardanoApplication . application
 
     -- Here we initialize all the features.
-    cardanoFeatures <- initializeAllFeatures cardanoConfiguration cardanoEnvironment
+    (cardanoFeatures, loggingLayer) <- initializeAllFeatures cardanoConfiguration cardanoEnvironment
```

Comment:
Right, with the addition that here we return the **top layer** that we need to run.